### ISSUE TITLE
KeyError: 'single_node'

### ISSUE THREAD

**Issue by Serikxx** [sentry-kubernetes/charts#594]
Hi. For some reason, an error occurs when installing in kubernetes. What can be wrong?

> kubectl logs sentry-snuba-db-init--1-81wdw -n sentry
Traceback (most recent call last):
  File "/usr/local/bin/snuba", line 33, in <module>
    sys.exit(load_entry_point('snuba', 'console_scripts', 'snuba')())
  File "/usr/local/bin/snuba", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/usr/local/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/src/snuba/snuba/cli/__init__.py", line 21, in <module>
    module = __import__(module_name, globals(), locals(), ["__name__"])
  File "/usr/src/snuba/snuba/cli/bootstrap.py", line 9, in <module>
    from snuba.migrations.connect import check_clickhouse_connections
  File "/usr/src/snuba/snuba/migrations/connect.py", line 7, in <module>
    from snuba.clusters.cluster import ClickhouseClientSettings, CLUSTERS
  File "/usr/src/snuba/snuba/clusters/cluster.py", line 330, in <module>
    CLUSTERS = [
  File "/usr/src/snuba/snuba/clusters/cluster.py", line 339, in <listcomp>
    single_node=cluster["single_node"],
KeyError: 'single_node'

in sentry-values.yml

> externalClickhouse:
host: "myhost"
  tcpPort: 9000
  httpPort: 8123
  username: user
  password: "password"
  database: db
  singleNode: true

> kubectl get all -n sentry
NAME                                    READY   STATUS             RESTARTS        AGE
pod/cm-acme-http-solver-747zr           1/1     Running            0               14m
pod/sentry-cron-754d8f78d4-zvr2j        1/1     Running            0               14m
pod/sentry-db-check--1-n659d            0/1     Completed          0               14m
pod/sentry-kafka-0                      1/1     Running            1 (13m ago)     14m
pod/sentry-kafka-1                      1/1     Running            1 (13m ago)     14m
pod/sentry-kafka-2                      1/1     Running            1 (14m ago)     14m
pod/sentry-snuba-api-7fdb795d4f-fkwrg   0/1     CrashLoopBackOff   7 (3m34s ago)   14m
pod/sentry-snuba-db-init--1-89brn       0/1     Error              0               12m
pod/sentry-snuba-db-init--1-czndr       0/1     Error              0               13m
pod/sentry-snuba-db-init--1-d8w2w       0/1     Error              0               13m
pod/sentry-snuba-db-init--1-h8n5m       0/1     Error              0               13m
pod/sentry-snuba-db-init--1-jq4jt       0/1     Error              0               4m25s
pod/sentry-snuba-db-init--1-s2plr       0/1     Error              0               9m45s
pod/sentry-snuba-db-init--1-vxtnh       0/1     Error              0               13m
pod/sentry-web-88d5f8786-bhcn5          1/1     Running            0               14m
pod/sentry-worker-7854fd8f-9hxsr        1/1     Running            0               14m
pod/sentry-worker-7854fd8f-r6tb8        1/1     Running            0               14m
pod/sentry-worker-7854fd8f-rpnkr        1/1     Running            0               14m
pod/sentry-zookeeper-0                  1/1     Running            0               14m

Although, when i use clickhouse: enabled: true without externalClickhouse everything is fine
clickhouse_version: 19.16.4.12
kubernetes version  1.22.7

**Comment by timfeirg** [comment]
I'm seeing the exact error on appVersion: 22.3.0, chart version: 14.0.2

**Comment by timfeirg** [comment]
fixed this KeyError with a little change to the configmap:

```
diff --git a/templates/configmap-snuba.yaml b/templates/configmap-snuba.yaml
index 93e6428..a065691 100644
--- a/templates/configmap-snuba.yaml
+++ b/templates/configmap-snuba.yaml
@@ -48,6 +48,8 @@ data:
         "single_node": False,
         "cluster_name": {{ include "sentry.clickhouse.cluster.name" . | quote }},
         "distributed_cluster_name": {{ include "sentry.clickhouse.cluster.name" . | quote }},
+        {{- else }}
+        "single_node": True,
         {{- end }}
       },
     ]
```

**Comment by Mokto** [comment]
This issue is stale because it has been open for 30 days with no activity.

**Comment by timfeirg** [comment]
the patch above has already been included in [cc4b4a6](https://github.com/sentry-kubernetes/charts/pull/588), I think this issue can be closed now @Mokto
